website/content/en/docs/setup/going-to-prod/hardening.md

```diff
@@ -0,0 +1,78 @@
+- **Disable core dumps.** A user who can force a core dump could access Vector’s in-flight data. Preventing core dumps is specific to your platform. On Linux setting the resource limit `RLIMIT_CORE` to `0` disables core dumps. In the systemd service unit file, setting `LimitCORE=0` will enforce this setting for the Vector service (this is done automatically when installing Vector through `apt` or `yum`).
+
+{{< info >}}
+Vector implements an affine type system via Rust that achieves memory safety and removes data from memory as soon as possible. Memory safety eliminates a class of memory-related security bugs, and the affine type system reduces exposure by only keeping data in memory when needed.
+{{< /info >}}
+
+### Securing the Vector Process
+
+#### Securing Vector’s Code
+
+{{< info >}}
+[Vector’s code is open-source](https://github.com/vectordotdev/vector), and the development process is secured as outlined in [Vector’s security policy](https://github.com/vectordotdev/vector/blob/master/SECURITY.md).
+{{< /info >}}
+
+#### Securing Vector’s Artifacts
+
+- **Download over encrypted channels.** Vector does not allow unencrypted downloads of its artifacts. All download channels require industry-standard TLS for all connections. When downloading Vector be sure to enable SSL verification (the default for most clients).
+
+#### Securing Vector’s Configuration
+
+- **Never use plain text secrets.** Never add plain text secrets to Vector’s configuration files.
+- **Restrict Vector’s configuration directory.** Vector’s configuration directory (i.e., `/etc/vector`) should be read restricted to Vector’s unprivileged service account and write restricted to your operational account used when deploying Vector.
+
+#### Securing Vector’s Runtime
+
+- **Do not run Vector as root.** Vector is designed to run via a dedicated service account. Never run Vector as root or an administrator account.
+- **Restrict the Vector service account.** Vector’s service account should not have the ability to overwrite Vector’s binary or configuration files (i.e., the `/etc/vector` directory). The only directory the Vector service account should write to is Vector’s data directory (i.e., `/var/lib/vector`).
+- **Upgrade Vector frequently.** Vector is actively developed by a team at Datadog and hundreds of contributors around the world. Releases can include important bug and security fixes. We recommend watching the [Vector repository](https://github.com/vectordotdev/vector) for releases, following the [Vector Twitter account](https://twitter.com/vectordotdev), or subscribing to the [Vector calendar](https://calendar.vector.dev) for release notifications.
+
+### Securing the Host
+
+- **Ensure Vector is the single tenant.** When deploying Vector as an aggregator it should be the single tenant on the machine. This prevents other processes from unknowingly interacting with the Vector process.
```

Comment:
I think this doesn't apply to containerized deployments. I realize this is straight from the Notion docs, but we may want to clarify this.